include/cudecomp.h

```diff
@@ -37,62 +37,67 @@
 #ifndef CUDECOMP_H
 #define CUDECOMP_H
 
-#include <string>
-
 #include <cuda_runtime.h>
 #include <mpi.h>
 
 #define CUDECOMP_MAJOR 0
 #define CUDECOMP_MINOR 4
 #define CUDECOMP_PATCH 2
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <stdbool.h>
+
 /**
  * @brief This enum lists the different available transpose backend options.
  */
-enum cudecompTransposeCommBackend_t {
+typedef enum cudecompTransposeCommBackend_t_enum {
```

Comment:
```suggestion
typedef enum {
```
Is the intermediate enum name necessary here? If not, please remove it. Otherwise, this causes some issues with the generation of the documentation.